boards/nxp/mimxrt1160_evk/Kconfig.mimxrt1160_evk

```diff
@@ -0,0 +1,13 @@
+#
+# Copyright 2024 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config BOARD_MIMXRT1160_EVK_RT1166_CM7
+	select SOC_PART_NUMBER_MIMXRT1166DVM6A
+	select SOC_MIMXRT1166_CM7
+
+config BOARD_MIMXRT1160_EVK_RT1166_CM4
+	select SOC_PART_NUMBER_MIMXRT1166DVM6A
+	select SOC_MIMXRT1166_CM4
```

Comment:
```suggestion
config BOARD_MIMXRT1160_EVK
	select SOC_PART_NUMBER_MIMXRT1166DVM6A
	select SOC_MIMXRT1166_CM7 if BOARD_MIMXRT1160_EVK_RT1166_CM7
	select SOC_MIMXRT1166_CM4 if BOARD_MIMXRT1160_EVK_RT1166_CM4
```

Reply:
I tried this and it does not build, because the SOC_MIMXRT1166_CMX kconfig is not selected, I dont know why

Reply:
I just tried it and it works for both cm7 and cm4?

Reply:
i'll try it again now that some other issues were resolved

Reply:
yeah it still didn't work:

```

warning: FLEXSPI_CONFIG_BLOCK_OFFSET (defined at soc/nxp/imxrt/rt11xx/Kconfig.defconfig:25,
soc/nxp/imxrt/Kconfig:63, soc/nxp/imxrt/Kconfig:130) was assigned the value '0x400' but got the
value ''. Check these unsatisfied dependencies: ((SOC_SERIES_IMXRT_11XX && SOC_FAMILY_IMXRT) ||
(NXP_IMXRT_BOOT_HEADER && SOC_FAMILY_IMXRT) || SOC_FAMILY_IMXRT) (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_FLEXSPI_CONFIG_BLOCK_OFFSET and/or look up
FLEXSPI_CONFIG_BLOCK_OFFSET in the menuconfig/guiconfig interface. The Application Development
Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual
might be helpful too.


warning: ARM_MPU (defined at arch/arm64/core/cortex_r/Kconfig:10, arch/arm/core/mpu/Kconfig:8) was
assigned the value 'y' but got the value 'n'. Check these unsatisfied dependencies: ((CPU_HAS_MPU &&
CPU_AARCH64_CORTEX_R && ARM64) || (CPU_HAS_MPU && ARM)) (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_ARM_MPU and/or look up ARM_MPU in the
menuconfig/guiconfig interface. The Application Development Primer, Setting Configuration Values,
and Kconfig - Tips and Best Practices sections of the manual might be helpful too.


warning: HW_STACK_PROTECTION (defined at arch/Kconfig:261) was assigned the value 'y' but got the
value 'n'. Check these unsatisfied dependencies: ARCH_HAS_STACK_PROTECTION (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_HW_STACK_PROTECTION and/or look up
HW_STACK_PROTECTION in the menuconfig/guiconfig interface. The Application Development Primer,
Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual might be
helpful too.


warning: UART_CONSOLE (defined at drivers/console/Kconfig:43) was assigned the value 'y' but got the
value 'n'. Check these unsatisfied dependencies: SERIAL_HAS_DRIVER (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_UART_CONSOLE and/or look up UART_CONSOLE in
the menuconfig/guiconfig interface. The Application Development Primer, Setting Configuration
Values, and Kconfig - Tips and Best Practices sections of the manual might be helpful too.

Parsing /home/declan/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/declan/zephyrproject/zephyr/boards/nxp/mimxrt1160_evk/mimxrt1160_evk_mimxrt1166_cm7_defconfig'
Merged configuration '/home/declan/zephyrproject/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/home/declan/zephyrproject/zephyr/build/zephyr/.config'
Kconfig header saved to '/home/declan/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h'
CMake Error at /home/declan/zephyrproject/zephyr/cmake/modules/arch_v2.cmake:36 (message):
  ARCH not defined.  Check that BOARD=mimxrt1160_evk, is selecting an
  appropriate SoC in Kconfig, SoC=, and that the SoC is selecting the correct
  architecture.
Call Stack (most recent call first):
  /home/declan/zephyrproject/zephyr/cmake/modules/zephyr_default.cmake:132 (include)
  /home/declan/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/declan/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
```

Reply:
it needs to match the board.yml name so
```
        select SOC_MIMXRT1166_CM7 if BOARD_MIMXRT1160_EVK_MIMXRT1166_CM7
        select SOC_MIMXRT1166_CM4 if BOARD_MIMXRT1160_EVK_MIMXRT1166_CM4
```

Reply:
@nordicjm apparently the issue was because I made the name of the kconfig BOARD_MIMXRT1170_EVK_MIMXRT1176 instead of BOARD_MIMXRT1170_EVK

I dont understand still why this would make the soc not get selected though, both of them are just arbitrary names of a new kconfig, no ?

Reply:
I guess it wasnt made default y, is there something special then about just BOARD_MIMXRT1170_EVK from the build system

Reply:
The Kconfigs are generated by the build system from board.yml